crates/forge_analyzer/src/interp.rs

```diff
@@ -676,9 +564,190 @@ impl<'cx, C: Runner<'cx>> Interp<'cx, C> {
+                .cloned()
+            {
+                // if there is an existing value...
+                match (existing_lval, rval_value) {
+                    // return unknown if either values are unknown
+                    (Value::Unknown, _)
+                    | (_, Value::Unknown)
+                    | (Value::Const(_), Value::Object(_))
+                    | (Value::Phi(_), Value::Object(_))
+                    | (Value::Object(_), Value::Phi(_))
+                    | (Value::Object(_), Value::Const(_)) => self.add_value_with_projection(
+                        defid_block,
+                        varid,
+                        Value::Unknown,
+                        projections,
+                    ),
+                    // push other const onto phi vec if either are const and phi
+                    (Value::Const(const_value), Value::Phi(phi_value))
+                    | (Value::Phi(phi_value), Value::Const(const_value)) => {
+                        let mut new_phi = phi_value;
+                        new_phi.push(const_value);
+                        self.add_value_with_projection(
+                            defid_block,
+                            varid,
+                            Value::Phi(new_phi),
+                            projections,
+                        )
+                    }
+                    // push consts into vec if both are consts
+                    (Value::Const(const_value1), Value::Const(const_value2)) => self
+                        .add_value_with_projection(
+                            defid_block,
+                            varid,
+                            Value::Phi(vec![const_value1, const_value2]),
+                            projections,
+                        ),
+                    (Value::Object(exist_var), Value::Object(new_var)) => {
+                        // store projection values that are transferred
+                        let mut projections_transferred = vec![];
+                        // tranfer all projection values from the new_var into the existing var
+                        for ((_, _, projections), value) in
+                            self.value_manager.varid_to_value.clone().range(
+                                (defid_block, new_var, ProjectionVec::new())
+                                    ..(defid_block, VarId(new_var.0 + 1), ProjectionVec::new()),
+                            )
+                        {
+                            projections_transferred.push(projections.clone());
+                            self.add_value_with_projection(
+                                defid_block,
+                                exist_var,
+                                value.clone(),
+                                projections.clone(),
+                            )
+                        }
+                        // clear any remaining values
+                        for ((_, _, _), value) in self
```

Comment:
Hmmm... I've spent some time trying to do this? Wouldn't we need to clone anyway since we are doing a query 2 times on a btreemap?

Reply:
Yeah, but you only need to clone the `new_var`'s `Value`s and `Projection`s. For instance:

```rust
    for ((_, _, _), value) in self.value_manager.varid_to_value.range_mut(
        (defid_block, exist_var, ProjectionVec::new())
            ..(defid_block, VarId(exist_var.0 + 1), ProjectionVec::new()),
    ) {
        *value = Value::Unknown
    }
    // tranfer all projection values from the new_var into the existing var
    let vals = self
        .value_manager
        .varid_to_value
        .range(
            (defid_block, new_var, ProjectionVec::new())
                ..(defid_block, VarId(new_var.0 + 1), ProjectionVec::new()),
        )
        .map(|((_, _, projections), value)| (projections.clone(), value.clone()))
        .collect_vec();
    for (projections, value) in vals {
        self.add_value_with_projection(defid_block, exist_var, value, projections)
    }
```

Reply:
When we add the values first, we call `self.get_farthest_obj()`. However, when we use the BTreeMap first, we don't check this instead assigning all of them to unknown. As a result, we cannot access the object via `self.get_farthest_object` since it is overwritten with unknown. But, I will implement the cloning only for the projections in the first one. Thank you!